Lesson45/gozde/recipe-app/src/example/Counter/__tests__/Counter.test.tsx

```diff
@@ -0,0 +1,86 @@
+    const countValue = screen.getByTestId("count-value");
+    const incrementButton = screen.getByTestId("increment-button");
+    const decrementButton = screen.getByTestId("decrement-button");
+
+    expect(countLabel).toBeInTheDocument();
+    expect(countValue).toBeInTheDocument();
+    expect(countValue).toHaveTextContent("0");
+    expect(incrementButton).toBeInTheDocument();
+    expect(decrementButton).toBeInTheDocument();
+  });
+
+  it("should render correctly with initial state specified", () => {
+    render(<Counter initialState={{ count: 5 }} />);
+
+    const countLabel = screen.getByTestId("count-label");
+    const countValue = screen.getByTestId("count-value");
+    const incrementButton = screen.getByTestId("increment-button");
+    const decrementButton = screen.getByTestId("decrement-button");
+
+    expect(countLabel).toBeInTheDocument();
+    expect(countValue).toBeInTheDocument();
+    expect(countValue).toHaveTextContent("5");
+    expect(incrementButton).toBeInTheDocument();
+    expect(decrementButton).toBeInTheDocument();
+  });
+
+  it("should increment count correctly when clicking on the + button", async () => {
+    render(<Counter />);
+
+    const countValue = screen.getByTestId("count-value");
+    const incrementButton = screen.getByTestId("increment-button");
+
+    expect(countValue).toHaveTextContent("0");
+    await userEvent.click(incrementButton);
+    expect(countValue).toHaveTextContent("1");
+
+    await userEvent.dblClick(incrementButton);
+    expect(countValue).toHaveTextContent("3");
+  });
+
+  it("should decrement count correctly when clicking on the - button", async () => {
+    render(<Counter />);
+
+    const countValue = screen.getByTestId("count-value");
+    const decrementButton = screen.getByTestId("decrement-button");
+
+    expect(countValue).toHaveTextContent("0");
+    await userEvent.click(decrementButton);
+    expect(countValue).toHaveTextContent("-1");
+
+    await userEvent.dblClick(decrementButton);
+    expect(countValue).toHaveTextContent("-3");
+  });
+});
+
+describe("counterReducer", () => {
+  it("should increment the count by 1 when the action type is INCREMENT", () => {
+    const initialState = { count: 0 };
+    const action = { type: CounterActionType.INCREMENT };
+    const newState = counterReducer(initialState, action);
+    expect(newState.count).toBe(1);
+  });
+
+  it("should decrement the count by 1 when the action type is DECREMENT", () => {
+    const initialState = { count: 0 };
+    const action = { type: CounterActionType.DECREMENT };
+    const newState = counterReducer(initialState, action);
+    expect(newState.count).toBe(-1);
+  });
+
+  it("should not increment or decrement the count when the action type does not match DECREMENT or INCREMENT", () => {
+    const initialState = { count: 0 };
+    const action = { type: "UNKNOWN" as unknown as CounterActionType }; // bilinmeyen eylem türü için ts deki hata bu şekilde çözüldü
```

Comment:
This fix is correct. The issue happened since CounterActionType does not include "UNKNOWN" value (it's unexpected value).